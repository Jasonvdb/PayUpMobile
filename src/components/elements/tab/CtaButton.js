import { StyleSheet, TouchableOpacity } from "react-native";
import Icon from "react-native-vector-icons/Ionicons";
import React from "react";
import PropTypes from "prop-types";
import * as Animatable from "react-native-animatable";
import LinearGradient from "react-native-linear-gradient";

import theme from "../../../config/theme";

const CtaButton = ({
  route,
  onPress,
  onLongPress,
  getAccessibilityLabel,
  isActive
}) => (
  <Animatable.View
    animation={isActive ? "bounceOut" : "bounceIn"}
    easing={"ease-in-out"}
    duration={500}
  >
    <TouchableOpacity
      onPress={() => onPress(route)}
      onLongPress={() => onLongPress(route)}
      accessibilityLabel={getAccessibilityLabel({
        route
      })}
    >
      <LinearGradient
        colors={theme.ctaButtonGradients}
        style={styles.ctaButton}
      >
        <Icon name={"ios-add"} size={35} color={"#FFF"}/>
      </LinearGradient>
    </TouchableOpacity>
  </Animatable.View>
);

CtaButton.propTypes = {
  route: PropTypes.object.isRequired,
  onPress: PropTypes.func.isRequired,
  onLongPress: PropTypes.func.isRequired,
  getAccessibilityLabel: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired
};

export default CtaButton;

const styles = StyleSheet.create({
  ctaButton: {
    top: -39,
    backgroundColor: theme.brand1,
    width: 56,
    height: 56,
    borderRadius: 35,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    paddingTop: 4,

    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.35,
    shadowRadius: 12,
    shadowColor: theme.brand1
  }
});
