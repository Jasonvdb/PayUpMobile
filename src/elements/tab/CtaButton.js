import { StyleSheet, TouchableOpacity, View } from "react-native";
import Icon from "react-native-vector-icons/Ionicons";
import React, { Fragment } from "react";
import PropTypes from "prop-types";
import * as Animatable from "react-native-animatable";

import theme from "../../config/theme";

const CtaButton = ({
	route,
	onPress,
	onLongPress,
	getAccessibilityLabel,
	isActive
}) => (
	<Animatable.View
		animation={isActive ? "bounceOut" : "bounceIn"}
		easing={"ease-in-out"}
		duration={500}
	>
		<TouchableOpacity
			style={styles.ctaButton}
			onPress={() => onPress(route)}
			onLongPress={() => onLongPress(route)}
			accessibilityLabel={getAccessibilityLabel({
				route
			})}
		>
			<Icon name={"ios-add"} size={35} color={"#FFF"}/>
		</TouchableOpacity>
	</Animatable.View>
);

CtaButton.propTypes = {
	route: PropTypes.object.isRequired,
	onPress: PropTypes.func.isRequired,
	onLongPress: PropTypes.func.isRequired,
	getAccessibilityLabel: PropTypes.func.isRequired,
	isActive: PropTypes.bool.isRequired
};

export default CtaButton;

const styles = StyleSheet.create({
	ctaButton: {
		top: -39,
		backgroundColor: theme.brand1,
		width: 56,
		height: 56,
		borderRadius: 35,
		display: "flex",
		justifyContent: "center",
		alignItems: "center",
		paddingTop: 4,

		shadowOffset: { width: 0, height: 4 },
		shadowOpacity: 0.35,
		shadowRadius: 12,
		shadowColor: theme.brand1
	}
});
