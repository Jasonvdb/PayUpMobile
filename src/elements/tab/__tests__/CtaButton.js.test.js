/**
 * @format
 */

import "react-native";
import React from "react";
import { TouchableOpacity } from "react-native";
import { shallow, mount } from "enzyme";

import CtaButton from "../CtaButton";

describe("CtaButton", () => {
	describe("Rendering", () => {
		it("should match to snapshot ", () => {
			const component = shallow(<CtaButton onPress={() => {}}/>);
			expect(component).toMatchSnapshot();
		});
	});

	describe("Interaction", () => {
		describe("onPress", () => {
			it("should call onPress", () => {
				const mockOnPress = jest.fn();
				const wrapper = shallow(<CtaButton onPress={mockOnPress}/>);
				wrapper
					.find(TouchableOpacity)
					.first()
					.props()
					.onPress();

				expect(mockOnPress).toHaveBeenCalled();
				expect(mockOnPress).toHaveBeenCalledTimes(1); // 5. checking return values
			});
		});
	});
});
