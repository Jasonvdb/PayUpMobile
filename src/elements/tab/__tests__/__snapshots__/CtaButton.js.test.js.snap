// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CtaButton Rendering should match to snapshot : Route active 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "opacity:1;transform:[object Object]",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "style": "top:-37px;background-color:#00239C;width:56px;height:56px;border-radius:35px;display:flex;justify-content:center;align-items:center;padding-top:4px;shadow-offset:[object Object];shadow-opacity:0.35px;shadow-radius:12px;shadow-color:#00239C;opacity:1",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "style": "0:[object Object];2:[object Object];3:[object Object]",
            },
            "children": Array [
              Object {
                "data": "",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "text",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
            },
          },
        ],
        "name": "view",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
    ],
    "name": "view",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`CtaButton Rendering should match to snapshot : Route not active 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "opacity:0;transform:[object Object]",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "style": "top:-37px;background-color:#00239C;width:56px;height:56px;border-radius:35px;display:flex;justify-content:center;align-items:center;padding-top:4px;shadow-offset:[object Object];shadow-opacity:0.35px;shadow-radius:12px;shadow-color:#00239C;opacity:1",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "style": "0:[object Object];2:[object Object];3:[object Object]",
            },
            "children": Array [
              Object {
                "data": "",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "text",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
            },
          },
        ],
        "name": "view",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
    ],
    "name": "view",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
